File: solutions/_203_remove_linked_list_elements/solution.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        dummy = ListNode(next=head)
        prev = dummy
        curr = head

        while curr:
            if curr.val == val:
                prev.next = curr.next
            else:
                prev = curr

            curr = curr.next

        return dummy.next


# https://leetcode.com/problems/remove-linked-list-elements/

File: solutions/_203_remove_linked_list_elements/Solution.java
package _203_remove_linked_list_elements;

class Solution {

  public ListNode removeElements(ListNode head, int val) {
    ListNode dummy = new ListNode(0);
    dummy.next = head;
    ListNode curr = dummy;

    while (curr.next != null) {
      if (curr.next.val == val) {
        curr.next = curr.next.next;
      } else {
        curr = curr.next;
      }
    }
    return dummy.next;
  }
}
/*
https://leetcode.com/problems/remove-linked-list-elements/
*/
